install.sh

```diff
@@ -62,134 +66,114 @@ function ask() {
 
-rm -f $STGDIR/repo/pacstallrepo.txt > /dev/null
-touch $STGDIR/repo/pacstallrepo.txt
-echo 'https://raw.githubusercontent.com/pacstall/pacstall-programs/master' > $STGDIR/repo/pacstallrepo.txt
+mkdir -p "$SRCDIR"
+chown "$(logname)" -R "$SRCDIR"
 
-fancy_message info "Pulling scripts from GitHub "
-for i in {error_log.sh,add-repo.sh,search.sh,download.sh,install-local.sh,upgrade.sh,remove.sh,update.sh,query-info.sh}; do
-	wget -q --show-progress -N https://raw.githubusercontent.com/pacstall/pacstall/master/misc/scripts/"$i" -P $STGDIR/scripts &
-done
+mkdir -p "$LOGDIR"
+mkdir -p "/var/log/pacstall/error_log"
+chown "$(logname)" -R "/var/log/pacstall/error_log"
+mkdir -p "/usr/share/man/man8"
+mkdir -p "/usr/share/bash-completion/completions"
 
-PID=$!
-i=1
-sp="/-\|"
-echo -n ' '
-while [ -d /proc/$PID ]
-do
-	sleep 0.1
-	printf "\b${sp:i++%${#sp}:1}"
-done
-echo ""
+rm -f "$STGDIR/repo/pacstallrepo.txt" > /dev/null
+touch "$STGDIR/repo/pacstallrepo.txt"
+echo "https://raw.githubusercontent.com/pacstall/pacstall-programs/master" > $STGDIR/repo/pacstallrepo.txt
 
-fancy_message info "pulling ${BLUE}pacstall${NC} from ${RED}https://raw.githubusercontent.com/pacstall/pacstall/master/pacstall${NC}"
+fancy_message info "Pulling scripts from GitHub"
+for i in {error_log.sh,add-repo.sh,search.sh,download.sh,install-local.sh,upgrade.sh,remove.sh,update.sh,query-info.sh}; do
+	wget -q --show-progress -N https://raw.githubusercontent.com/pacstall/pacstall/master/misc/scripts/"$i" -P "$STGDIR/scripts" &
+done
 
-sudo wget -q --show-progress --progress=bar:force -O /bin/pacstall https://raw.githubusercontent.com/pacstall/pacstall/master/pacstall &
-wget -q --show-progress --progress=bar:force -O /usr/share/man/man8/pacstall.8.gz https://raw.githubusercontent.com/pacstall/pacstall/master/misc/pacstall.8.gz &
-sudo wget -q --show-progress --progress=bar:force -O /usr/share/bash-completion/completions/pacstall https://raw.githubusercontent.com/pacstall/pacstall/master/misc/completion/bash &
+wget -q --show-progress --progress=bar:force -O "/bin/pacstall" "https://raw.githubusercontent.com/pacstall/pacstall/master/pacstall" &
+wget -q --show-progress --progress=bar:force -O "/usr/share/man/man8/pacstall.8.gz" "https://raw.githubusercontent.com/pacstall/pacstall/master/misc/pacstall.8.gz" &
 
-if command -v fish &>/dev/null; then
-	sudo wget -q --show-progress --progress=bar:force -O /usr/share/fish/vendor_completions.d/pacstall.fish https://raw.githubusercontent.com/pacstall/pacstall/master/misc/completion/fish &
-fi
+wget -q --show-progress --progress=bar:force -O "/usr/share/bash-completion/completions/pacstall" "https://raw.githubusercontent.com/pacstall/pacstall/master/misc/completion/bash" &
+wget -q --show-progress --progress=bar:force -O "/usr/share/fish/vendor_completions.d/pacstall.fish" "https://raw.githubusercontent.com/pacstall/pacstall/master/misc/completion/fish" &
 
 wait
 
-sudo chmod +x /bin/pacstall
+chmod +x "/bin/pacstall"
```

Comment:
This will fail in a non sudo environment

Reply:
The script is running using `sudo` everytime. See the installation snippet
```bash
sudo bash -c "$(curl -fsSL https://git.io/JsADh || wget -q https://git.io/JsADh -O -)"
```

Reply:
> The script is running using `sudo` everytime. See the installation snippet
> 
> ```shell
> sudo bash -c "$(curl -fsSL https://git.io/JsADh || wget -q https://git.io/JsADh -O -)"
> ```

small 🧠 moment